### PR TITLE
live gw fix

### PR DESCRIPTION
**Pull request by johnmadden86**

### CODE REVIEW

**Comment by johnmadden86** [comment]
yes, double up not required. live fixture info was in the /live endpoint in last season's api

**Comment by amosbastian** [comment]
If you remove the last commit I think it's good to get merged. Travis CI fails because they do not provide the environment variables needed to log in to untrusted builds, triggered by pull requests from another repository, so I'll test it locally.

For future's reference it would be cool if you could create an issue with the problem beforehand / include a description so I know what you are trying to fix :grin:

**Comment by johnmadden86** [comment]
Cool, thanks for the feedback.
Getting a bit more used to it now.
Haven't ever really used git properly before.

File: fpl/models/fixture.py [hunk 1]
```diff
@@ -2,14 +2,15 @@
 from .player import Player
 from datetime import datetime
```
Comment (amosbastian) [inline fpl/models/fixture.py]: Can remove this since it's not used anymore.

**Comment by amosbastian** [comment]
I've been busy with irl stuff (job interviews and such) so haven't had time to check out the changes, sorry! Let me know if you are done adding stuff and I can review everything and eventually merge it. Cheers!

**Comment by johnmadden86** [comment]
Hope to keep contributing, though like yourself, it might be a bit sporadic.
I'm wondering if the live data methods would be better in the Gameweek class.

**Comment by amosbastian** [comment]
Ok I just pulled this to my computer to see how it's going and see that a lot of tests are now failing. A couple of tips:

1. Limit your pull request to one feature. You've now changed a lot of things in different places, such as using `getattr` in `fpl.get_players` or `fpl.get_team` (which breaks them), when this pull request was originally meant as a fix for just `fpl.get_gameweek` iirc.
2. Following from the above: create an issue for each thing you want to change so we can discuss if it's necessary / a good idea beforehand. It's not nice for you if you put a lot of hard work into something and then I don't agree with it being added for example.
3. Run tests after you change stuff to make sure you aren't breaking anything. For example now there are 9 failing tests and 4 that have an error.

I'm not that sure what the next step should be, but maybe you could create a new pull request with just the original live gameweek fix and create issues for all the other stuff you added and we can discuss them?

**Comment by johnmadden86** [comment]
Hi,

I made a couple of small changes and now the unit tests are passing (except
for the H2H league one).

Regarding the other stuff, I don't really have any experience of working
collaboratively or using Git properly.
I've only really used it in a similar way you'd use dropbox.

I did a lot this stuff last year in nodejs, but it was just a script for
myself really.
I don't know if other people when they viewed it, would be able to utilise
it or contribute to it.

I thought you'd be able to view and merge whichever of my commits you liked?
Git was putting them all into to one pull request, do I need a separate
branch for each change I'm making?

I'll go back over my commits and open up issues.
It's certainly be a good idea to get consensus on what return type people
wanted and how the data is to be structured.

**Comment by amosbastian** [comment]
Basically it's normal to create a new branch for each feature that you want to add. Once you create a pull request from a branch then each commit that you make afterwards is automatically added to it. Currently you've made a lot of changes that should've probably each been their own branch / pull request.

**Comment by johnmadden86** [comment]
Ok, I might make a new fork, or rebase and commit the changes in separate
branches

**Pull request closed without merging**